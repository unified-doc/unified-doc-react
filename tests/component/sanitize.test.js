import React from 'react';
import renderer from 'react-test-renderer';

import Doc from '../../lib/component';

const htmlContent = '<div classname="red" style="background: red;">text</div>';

describe('sanitize', () => {
  it('sanitizes html by default', () => {
    const tree = renderer
      .create(<Doc options={{ content: htmlContent, filename: 'doc.html' }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies custom sanitize schema', () => {
    const sanitizeSchema = { attributes: { '*': ['style'] } };
    const tree = renderer
      .create(
        <Doc
          options={{
            content: htmlContent,
            filename: 'doc.html',
            sanitizeSchema,
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('does not apply sanitize schema', () => {
    const tree = renderer
      .create(
        <Doc
          options={{
            content: htmlContent,
            filename: 'doc.html',
            sanitizeSchema: null,
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
