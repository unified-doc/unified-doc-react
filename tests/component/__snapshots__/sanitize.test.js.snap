// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanitize applies custom sanitize schema 1`] = `
<div>
  <div>
    <div
      style={
        Object {
          "background": "red",
        }
      }
    >
      text
    </div>
  </div>
</div>
`;

exports[`sanitize does not apply sanitize schema 1`] = `
<div>
  <div>
    <html>
      <head />
      <body>
        <div
          className="red"
          style={
            Object {
              "background": "red",
            }
          }
        >
          text
        </div>
      </body>
    </html>
  </div>
</div>
`;

exports[`sanitize sanitizes html by default 1`] = `
<div>
  <div>
    <div>
      text
    </div>
  </div>
</div>
`;
